src/Microsoft.Diagnostics.Monitoring.EventPipe/DistributedTraces/TraceEventExtensions.cs

```diff
@@ -0,0 +1,223 @@
+                {
+                    string? key = arg["Key"] as string;
+                    object value = arg["Value"];
+
+                    switch (key)
+                    {
+                        case "TraceId":
+                            string? traceIdValue = value as string;
+                            if (!string.IsNullOrEmpty(traceIdValue)
+                                && traceIdValue != DefaultTraceId)
+                            {
+                                traceId = ActivityTraceId.CreateFromString(traceIdValue);
+                            }
+                            break;
+                        case "SpanId":
+                            string? spanIdValue = value as string;
+                            if (!string.IsNullOrEmpty(spanIdValue)
+                                && spanIdValue != DefaultSpanId)
+                            {
+                                spanId = ActivitySpanId.CreateFromString(spanIdValue);
+                            }
+                            break;
+                        case "ParentSpanId":
+                            string? parentSpanIdValue = value as string;
+                            if (!string.IsNullOrEmpty(parentSpanIdValue)
+                                && parentSpanIdValue != DefaultSpanId)
+                            {
+                                parentSpanId = ActivitySpanId.CreateFromString(parentSpanIdValue);
+                            }
+                            break;
+                        case "ActivityTraceFlags":
+                            if (value is string traceFlagsValue)
+                            {
+                                traceFlags = (ActivityTraceFlags)Enum.Parse(typeof(ActivityTraceFlags), traceFlagsValue);
+                            }
+                            break;
+                        case "TraceStateString":
+                            traceState = value as string;
+                            break;
+                        case "Kind":
+                            if (value is string kindValue)
+                            {
+                                kind = (ActivityKind)Enum.Parse(typeof(ActivityKind), kindValue);
+                            }
+                            break;
+                        case "DisplayName":
+                            string? displayNameValue = value as string;
+                            if (!string.IsNullOrEmpty(displayNameValue)
+                                && displayNameValue != activityName)
+                            {
+                                displayName = displayNameValue;
+                            }
+
+                            break;
+                        case "StartTimeTicks":
+                            if (value is string startTimeUtcValue)
+                            {
+                                startTimeUtc = DateTime.SpecifyKind(new DateTime(long.Parse(startTimeUtcValue)), DateTimeKind.Utc);
```

Comment:
Nit: Can't you construct the DateTime directly from ticks and DateTimeKind.Utc?
i.e.
```csharp
startTimeUtc = new DateTime(long.Parse(startTimeUtcValue), DateTimeKind.Utc);
```
?